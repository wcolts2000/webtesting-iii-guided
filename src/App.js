import React, { Component } from "react";
import "./App.css";

import Speaker from "./components/Speaker";

class App extends Component {
  state = { message: "nothing to say" };

  render() {
    return (
      <div className="App">
        <Speaker message={this.state.message} speak={this.speak} />
      </div>
    );
  }

  speak = () => {
    this.setState({ message: "you are not mocking me" });
  };
}

export default App;
