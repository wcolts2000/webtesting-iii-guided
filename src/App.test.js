import React from "react";
import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render, fireEvent } from "react-testing-library";
import "jest-dom/extend-expect";

import App from "./App";

describe("<App />", () => {
  it.skip("runs the tests", () => {
    expect(true).toBe(true);
  });
  // 2. write this test
  it("matches snapshot", () => {
    const tree = renderer.create(<App />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
  // it.only("runs only one test", () => {
  //   expect(false).toBe(false);
  // });
});

describe("mocking", () => {
  it("is mocking me", () => {
    const mock = jest.fn();

    const actual = mock("smile");

    expect(actual).toBeUndefined();
    expect(mock).toHaveBeenCalled();
    expect(mock).toHaveBeenCalledTimes(1);
  });
  it("controls the mock", () => {
    const mock = jest.fn(() => "hello");

    const actual = mock("smile");

    expect(actual).toBe("hello");
    expect(mock).toHaveBeenCalled();
    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith("smile");
  });
});

describe("speak()", () => {
  it("updates the message when speak button is clicked", () => {
    const { getByText, queryByText } = render(<App />);

    expect(queryByText(/not mocking me/i)).toBeFalsy();

    const button = getByText(/speak/i);
    fireEvent.click(button);

    expect(queryByText(/not mocking me/i)).toBeTruthy();
  });
});
