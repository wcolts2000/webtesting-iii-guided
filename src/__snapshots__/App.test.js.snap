// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Speak
  </button>
  <div>
    nothing to say
  </div>
</div>
`;
